hikari/impl/rest.py

```diff
@@ -1593,15 +1592,18 @@ async def delete_messages(
             # I am just gonna invoke these sequentially instead.
             try:
                 if len(pending) == 1:
-                    message = pending.popleft()
+                    message = pending[0]
                     await self.delete_message(channel, message)
                     deleted.append(message)
+
                 else:
                     body = data_binding.JSONObjectBuilder()
-                    chunk = [pending.popleft() for _ in range(min(100, len(pending)))]
+                    chunk = pending[: min(100, len(pending))]
```

Comment:
This is min statement unnecessary, `list[:100]` will give you up to 100 entries but less if there's not 100 entries present
e.g.
```py
>>> [1,2,3,4][:100]
[1, 2, 3, 4]
```